### PR TITLE
Param

### PR DESCRIPTION
**Pull request by sazzad16**

### CODE REVIEW

File: src/main/java/com/redislabs/redistimeseries/CreateParams.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    return this;
+  }
+
+  public CreateParams label(String label, String value) {
+    if (this.labels == null) this.labels = new HashMap<>();
+    this.labels.put(label, value);
+    return this;
+  }
+
+  public CreateParams labels(String... labels) {
```
Comment (gkorland) [inline src/main/java/com/redislabs/redistimeseries/CreateParams.java:50]: @sazzad16 I think it worth a JDoc to explain how you "parse" the labels

File: src/main/java/com/redislabs/redistimeseries/CreateParams.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    final int len = labels.length;
+    if ((len & 1) != 0) {
+      throw new IllegalArgumentException();
```
Comment (gkorland) [inline src/main/java/com/redislabs/redistimeseries/CreateParams.java:53]: Worth some error text in the exception so the user will understand his mistake

File: src/main/java/com/redislabs/redistimeseries/CreateParams.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    }
+    return labels(map);
+  }
+
+  public CreateParams labels(Map<String, String> labels) {
+    this.labels = labels;
+    return this;
+  }
+
+  public byte[][] getCreateByteParams(String key) {
```
Comment (gkorland) [inline src/main/java/com/redislabs/redistimeseries/CreateParams.java:67]: I think it's better to drop getCreateByteParams and getAddByteParams, and make addOptionalParams public.
The caller should pass the list and this class shouldn't not deal with serializing external arguments like key, time, value

File: src/test/java/com/redislabs/redistimeseries/RedisTimeSeriesTest.java [hunk 4]
```diff
@@ -298,8 +298,8 @@ public void del() {
     assertTrue(client.create("ts-del", 10000L));
     assertTrue(client.del("ts-del", 0, 1));
 
-    assertEquals(1000L, client.add("ts-del", 1000L, 1.1, 10000, null));
-    assertEquals(2000L, client.add("ts-del", 2000L, 0.9, null));
+    assertEquals(1000L, client.add("ts-del", 1000L, 1.1, 10000));
```
Comment (gkorland) [inline src/test/java/com/redislabs/redistimeseries/RedisTimeSeriesTest.java:301]: Why did you change these tests?

**Review by gkorland — CHANGES_REQUESTED** [review]

File: src/test/java/com/redislabs/redistimeseries/RedisTimeSeriesTest.java (hunk 4 above)
Comment (sazzad16) [reply src/test/java/com/redislabs/redistimeseries/RedisTimeSeriesTest.java:301]: @gkorland I just added these in #60 mostly copying from `testAdd()` test. Due to change in `add(...)` method, now we have to type-cast the null. Instead of doing that, I used simplified versions because actual `add(...)` is being (and should be) tested in `testAdd()` and we should be more concerned about testing `del(...)` command here, not `add(...)`.

File: src/main/java/com/redislabs/redistimeseries/CreateParams.java (hunk 1 above)
Comment (sazzad16) [reply src/main/java/com/redislabs/redistimeseries/CreateParams.java]: @gkorland I added this method in my previous approach where I was calling new `add` method from all existing `add` methods. This is not required in this approach.

File: src/main/java/com/redislabs/redistimeseries/CreateParams.java (hunk 3 above)
Comment (sazzad16) [reply src/main/java/com/redislabs/redistimeseries/CreateParams.java:48]: @gkorland That approach would require more intermediate lists/arrays. For example, creating a list/array in `addOptionalParams` with all the optional arguments and then creating another list/array unifying key, time, value with previous one.

Comment (gkorland) [reply src/main/java/com/redislabs/redistimeseries/CreateParams.java:48]: Why? exactly like you pass `addOptionalParams(List<byte[]> params)` is getting a list now, it can get the list in `add` with the key, timestamp and value already there.

Comment (sazzad16) [reply src/main/java/com/redislabs/redistimeseries/CreateParams.java:48]: @gkorland Let me know if prefer #66

Comment (gkorland) [reply src/main/java/com/redislabs/redistimeseries/CreateParams.java:48]: yes I think it's the right spread of responsibility
